### ISSUE TITLE
Nested manual trigger doesn't work

### ISSUE THREAD

**Issue by linupi** [myclabs/jquery.confirm#54]
using manually triggered confirms in a chain doesn't work. E.g. in 

```javascript
 $.confirm({
    text: "Question 1?",
    confirm: function() {
        $.confirm({
          text: "Question 2?",
          confirm: function() {
            // do something 
          },
          cancel: function() {
            // do something else
          }
        });
    },
    cancel: function() {
        // nothing to do
    }
  })
```

the 'inner' confirm dialog will not show up at all.

**Comment by JohannMarx** [comment]
Same issue here, I have two confirmation boxes that have to display after one another and the second one is not displayed at all.

	dialog1()
		.then(function (confirmed) {
		    if (!confirmed)
		        return false;

		        dialog2()
		            .then(function(result) {
		                if (result) {
		                    return function1;
		                } else {
		                    return function2;
		                }
		            });

		    return false;
		});

                    return false;
                });

Any ideas on how we can get this to work?

**Comment by JohannMarx** [comment]
I managed to sort it out, @linupi. It might be a bit of a hack, but it works. I had to add code to hide the "confirmation-modal" and "modal-backdrop" divs once the confirmation is closed. 

I'm sharing the solution here in case you can use it as well and it might be a bit specific to my requirement, but hopefully it helps. I am using HotTowelAngular for the site, so the $q.defer is part of that.

	function showConfirmation(message) {
	    var deferred = $q.defer();

	    hideConfirmation();

	    if (!message)
	        message = "Confirm selected action?";

	    $.confirm({
	        title: "Please Confirm",
	        text: message,
	        confirm: function () {
	            deferred.resolve(true);
	        },
	        cancel: function () {
	            deferred.resolve(false);
	            hideConfirmation(true);
	        },
	        confirmButton: "Yes",
	        cancelButton: "No",
	        confirmButtonClass: "btn-success",
	        cancelButtonClass: "btn-danger",
	        dialogClass: "modal-dialog modal-sm confirmation"
	    });

	    return deferred.promise;
	}

	function hideConfirmation(kill) {
	    var currentDisplayedDialog = $(".confirmation-modal");

	    if (currentDisplayedDialog && currentDisplayedDialog.length > 0) {
	        currentDisplayedDialog.remove();
	    }

	    if (kill) {
	        var currentDialogBackdrop = $(".modal-backdrop");

	        if (currentDialogBackdrop && currentDialogBackdrop.length > 0) {
	            currentDialogBackdrop.remove();
	        }
	    }
	}

**Comment by Deuteu** [comment]
Due to [jquery.confirm.js#L45](https://github.com/myclabs/jquery.confirm/blob/master/jquery.confirm.js#L45)
```javascript
if ($('.confirmation-modal').length > 0)
            return;
```
You can't open a second `.confirmation-modal` before you close the first one so you should trigger the second `.confirmation-modal` after with [bootstrap modal events](https://getbootstrap.com/javascript/#modals-events)

There is an example :
```javascript
$.confirm({
    text: "Question 1?",
    confirm: function(button) {
        $('.confirmation-modal').on('hidden.bs.modal', function () {
            $.confirm({
              text: "Question 2?",
              confirm: function() {
                // do something 
              },
              cancel: function() {
                // do something else
              }
            });
        });
    },
    cancel: function() {
        // nothing to do
    }
})
```

**Comment by mnapoli** [comment]
I'm 👍  to allow nested confirms if it can be done ;)

**Comment by Deuteu** [comment]
I needed to use #73 for `nested-option` branch so waiting to deal with this one and will update my branch and PR it.

Moreover, forgot to update Readme for this one too. ;)
